Formula/python@3.9.rb

```diff
@@ -309,6 +309,23 @@ def post_install
       (HOMEBREW_PREFIX/"bin").install_symlink bin/e
     end
 
+    # Replace bundled setuptools/pip with our own
+    rm Dir["#{lib_cellar}/ensurepip/_bundled/{setuptools,pip}-*.whl"]
+    system bin/"pip3", "wheel", "--wheel-dir=#{lib_cellar}/ensurepip/_bundled",
+           libexec/"setuptools", libexec/"pip"
+
+    # Patch ensurepip to bootstrap our updated versions of setuptools/pip
+    setuptools_whl = Dir["#{lib_cellar}/ensurepip/_bundled/setuptools-*.whl"][0]
```

Comment:
The array indexing here is a little cryptic, but perhaps could just be clarified in a future PR.

Reply:
Maybe just `.first`?

Reply:
`.first` is an improvement. That array should only ever have one entry in it. If there's some more idiomatic way to extract the filename of the only file in that directory, it would probably make it more readable.

(I wrote that array indexing tidbit right there back in #67718 before we decided to split this off into its own PR.)

Reply:
I should also mention - another option is to extract the version information from the `url`s for the `resource` blocks for `setuptools` and `pip`.

Reply:
Parsing it from the URLs might be better. In fact, it seems like `resource` blocks already do this? https://rubydoc.brew.sh/Resource.html

Except that the `version` attribute appears to be private... Is it possible to expose this publicly? Seems like it could be useful elsewhere.